integration_tests/src/fixture_generator/generator.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! This module contains the struct definitions of the tool output and main
+//! fixture generation logic.
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Write;
+
+use akd::directory::Directory;
+use akd::storage::types::{AkdLabel, AkdValue, DbRecord};
+use akd::storage::StorageUtil;
+use clap::Parser;
+use rand::{rngs::OsRng, Rng};
+use serde::{Deserialize, Serialize};
+use winter_crypto::hashers::Blake3_256;
+use winter_math::fields::f128::BaseElement;
+
+use crate::fixture_generator::parser::Args;
+use crate::fixture_generator::writer::yaml::YamlWriter;
+use crate::fixture_generator::writer::Writer;
+
+type Blake3 = Blake3_256<BaseElement>;
+
+/// Directory state comprises all database records at a particular epoch.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct State {
+    pub epoch: u32,
+    pub records: Vec<DbRecord>,
+}
+
+/// Delta comprises all updates published to the directory to advance to an
```

Comment:
Clarify "key" updates? Also in the arg description too.


Should we also consider having a delta of updated **nodes** so that multiple fixtures can be generated/advanced to the next state without performing the actual computation? Let me know if you'd like to discuss this more.